extensions/canvasMedical/v1/actions/createClaim/config/fields.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  DateOnlySchema,
+  FieldType,
+  type Field,
+} from '@awell-health/extensions-core'
+import { z, type ZodTypeAny } from 'zod'
+
+export const fields = {
+  status: {
+    id: 'status',
+    label: 'Status',
+    description:
+      'Accepts a value representing the status compliant with Canvas Medical.',
+    type: FieldType.STRING,
+    required: true,
+  },
+  type: {
+    id: 'type',
+    label: 'Type',
+    description:
+      'Accepts a JSON object representing the type compliant with Canvas Medical.',
+    type: FieldType.JSON,
+    required: true,
+  },
+  patientId: {
+    id: 'patientId',
+    label: 'Patient ID',
+    description: `A patient's unique ID.`,
+    type: FieldType.STRING,
+    required: true,
+  },
+  created: {
+    id: 'created',
+    label: 'Created',
+    description:
+      'The field indicating when this claim resource was created, accepting a date in the YYYY-MM-DD format.',
+    type: FieldType.DATE,
+    required: true,
+  },
+  provider: {
+    id: 'provider',
+    label: 'Provider',
+    description:
+      'The field related to determining personnel resources for services in the statement accepts a JSON object in accordance with the Canvas Medical format.',
+    type: FieldType.JSON,
+    required: true,
+  },
+  supportingInfo: {
+    id: 'supportingInfo',
+    label: 'Supporting Info',
+    description:
+      'Accepts a value representing the supporting info compliant with Canvas Medical.',
+    type: FieldType.JSON,
+    required: false,
+  },
+  diagnosis: {
+    id: 'diagnosis',
+    label: 'Diagnosis',
+    description:
+      'The field represents the list will create the Assessments in Canvas for this claim and accepts an array of JSON objects.',
+    type: FieldType.JSON,
+    required: true,
+  },
+  insurance: {
+    id: 'insurance',
+    label: 'Insurance',
+    description:
+      'The field represents the list of elements that defines what coverages are to be used when adjudicating the claim and accepts an array of JSON objects.',
+    type: FieldType.JSON,
+    required: true,
+  },
+  item: {
+    id: 'item',
+    label: 'Item',
+    description:
+      'The field represents the list of service charges to be used in the claim and accepts an array of JSON objects.',
+    type: FieldType.JSON,
+    required: true,
+  },
+} satisfies Record<string, Field>
+
+export const fieldsValidationSchema = z.object({
+  status: z.string(),
+  type: z.record(z.string(), z.any()),
+  patientId: z.string(),
+  created: DateOnlySchema,
+  provider: z.record(z.string(), z.any()),
+  supportingInfo: z.array(z.any()).optional(),
+  diagnosis: z.array(z.any()),
+  insurance: z.array(z.any()),
+  item: z.array(z.any()),
```

Comment:
I believe that won't work. Look at e.g. `extensions/sendbird/v1/actions/updateCustomerCustomFields/config/fields.ts` to see how to validate JSON fields (there is json validator in `extensions/sendbird/v1/validation.ts`). I think `json` fields are not automatically parsed into objects/arrays, at least that was the case a few weeks ago.

cc @rahulkeerthi to confirm

Reply:
Well! Thanks for pointing out this problem. I will try to correct it.

Reply:
Yes, that's right @michal-grzelak